### PR TITLE
role update: search all assigned scopes for existing role definitions

### PR DESCRIPTION
**Pull request by yugangw-msft**
- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yugangw-msft** [comment]
@adewaleo, could you please review?

File: src/azure-cli/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -109,19 +109,22 @@ def delete_role_definition(cmd, name, resource_group_name=None, scope=None,
...
+def _search_role_definitions(cli_ctx, definitions_client, name, scopes, custom_role_only=False):
+    for scope in scopes:
+        roles = list(definitions_client.list(scope))
+        worker = MultiAPIAdaptor(cli_ctx)
+        if name:
+            roles = [r for r in roles if r.name == name or worker.get_role_property(r, 'role_name') == name]
+        if custom_role_only:
+            roles = [r for r in roles if worker.get_role_property(r, 'role_type') == 'CustomRole']
+        if roles:
+            return roles
```
Comment (adewaleo) [inline src/azure-cli/azure/cli/command_modules/role/custom.py]: Do we want to return roles and break out of the for loop immediately or search all scopes and keep updating roles?

Comment (yugangw-msft) [reply src/azure-cli/azure/cli/command_modules/role/custom.py]: The old behavior is to only update one, which is the main scenario which I would like to maintain.

Comment (yugangw-msft) [reply src/azure-cli/azure/cli/command_modules/role/custom.py]: We don't need a history entry since it is a bug fix towards a pretty edge scenario

**Pull request merged**